### DISCUSSION TITLE
svn support for `affected` related commands such as format

### DISCUSSION THREAD

**Discussion by sacru2red** [nrwl/nx#28612] [Feature Requests]
### Current Behavior

I'am on svn not git.

and "nx format" command throw me error

I tried with "--uncommitted", "--untracked" ... options

![Image](https://github.com/user-attachments/assets/25b7da30-f10d-4d0b-8bbd-7bac65bc5922)

```
error: Could not access 'master'

 NX   Command failed: git diff --name-only --no-renames --relative "master" "HEAD"
error: Could not access 'master'


Defaulting to all files pattern: "."

[error] No files matching the pattern were found: "nx.json".
[error] No files matching the pattern were found: "tsconfig.base.json".
```

### Expected Behavior

no errors

### Steps to Reproduce

1. start on Existing Repository without git
2. run format

### Nx Report

Node           : 20.17.0
OS             : win32-x64
Native Target  : x86_64-windows
npm            : 10.8.3

nx             : 20.0.2
@nx/js         : 20.0.2
@nx/jest       : 20.0.2
@nx/eslint     : 20.0.2
@nx/workspace  : 20.0.2
@nx/devkit     : 20.0.2
@nx/next       : 20.0.2
@nx/react      : 20.0.2
@nx/web        : 20.0.2
@nx/webpack    : 20.0.2
typescript     : 5.5.4
---------------------------------------
Registered Plugins:
@nx/jest/plugin
@nx/next/plugin

### Operating System

- [ ] macOS
- [ ] Linux
- [x] Windows
- [ ] Other (Please specify)

**Comment by JamesHenry** [comment]
@sacru2red Thank you but lack of svn support isn't a bug, it's just not something that has ever been requested and it hasn't been prioritized on our roadmap. Our commands which leverage `affected` information (of which `format` is one) all rely on git so they won't work as expected for you for that reason. You could run `prettier` yourself instead of going through `nx format` and derive the relevant files using corresponding svn logic.

I will transfer this to a feature discussion as svn support is a much broader topic.

**Reply by sacru2red** [reply]
well. What I want is no errors, not support for svn. @JamesHenry

**Reply by JamesHenry** [reply]
I understand, but as you are not using the supported version control system, you will get errors. Getting rid of the errors would involve expanding the scope of support beyond git which is a feature we’re not currently working on. It’s really semantics whether you want to call it a bug or a feature request, but the long and short of it is that you cannot run commands that rely on git to function right now if you are not also using git.

My suggestion on what you can do today on your side to get the same experience as nx format still stands as well
